### PR TITLE
tkt-35769: Interfaces Management v2

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/network.py [hunk 1]
```diff
@@ -313,20 +389,636 @@ def iface_extend(self, iface):
...
+                            ' on members interfaces.',
+                        )
+        elif itype == 'LINK_AGGREGATION':
+            if 'name' in data and not (
+                data['name'].startswith('lagg') and data['name'][4:].isdigit()
+            ):
+                verrors.add(
+                    f'{schema_name}.name',
+                    (
+                        'Link aggregation interface must start with "lagg" followed by an unique'
```
[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/network.py]: Line 713

File: src/middlewared/middlewared/plugins/network.py [hunk 2]
```diff
@@ -313,20 +389,636 @@ def iface_extend(self, iface):
...
+            await self.middleware.call('datastore.insert', 'network.vlan', i)
+
+        for i in self._original_datastores['lagg']:
+            await self.middleware.call('datastore.insert', 'network.lagginterface', i)
+
+        for i in self._original_datastores['laggmembers']:
+            await self.middleware.call('datastore.insert', 'network.lagginterfacemembers', i)
+
+    @accepts()
+    async def has_pending_changes(self):
```
[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/network.py]: Well, they are not applied in some sense.

The changes are in database but they are pending to be applied and become effective.

The way I see it unapproved means changes are in effect but not approved.

[the remark this answers is not in the document]
Comment (themylogin) [reply src/middlewared/middlewared/plugins/network.py]: OK, I see. What is the method that will tell us if we must approve our current changes or they would be rolled back in XX seconds then?

[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/network.py]: Good point, just added `interfaces.checkin_waiting`

**Pull request merged**